tests/test_ess_csc.py

```diff
@@ -75,71 +80,76 @@ async def test_bin_script(self):
         async with self.make_csc(
             initial_state=salobj.State.STANDBY,
             config_dir=None,
             simulation_mode=1,
-            local_mode=True,
         ):
+            self.csc.port = self.port
+            self.assertFalse(self.socket_server.connected)
             await salobj.set_summary_state(
                 remote=self.remote, state=salobj.State.ENABLED
             )
+            self.assertTrue(self.socket_server.connected)
+
+            # Make sure that the mock sensor outputs data for a disconnected
+            # channel.
+            self.socket_server.command_handler._devices[
+                0
+            ]._disconnected_channel = self.disconnected_channel
+
+            # Make sure that the mock sensor outputs truncated data.
+            self.socket_server.command_handler._devices[
+                0
+            ]._missed_channels = self.missed_channels
+
             await self.validate_telemetry()
 
-    async def test_receive_telemetry_with_nan(self):
-        logging.info("test_receive_telemetry")
-        async with self.make_csc(
-            initial_state=salobj.State.STANDBY,
-            config_dir=None,
-            simulation_mode=1,
-            local_mode=True,
-        ):
-            nan_channel = 1
-            self.csc.nan_channel = nan_channel
-            await salobj.set_summary_state(
-                remote=self.remote, state=salobj.State.ENABLED
-            )
-            await self.validate_telemetry(nan_channel=nan_channel)
+            # Reset self.missed_channels and read again. The data should not be
+            # truncated anymore.
+            self.missed_channels = 0
 
-    async def test_remote_mode(self):
-        logging.info("test_remote_mode")
-        async with self.make_csc(
-            initial_state=salobj.State.STANDBY,
-            config_dir=None,
-            simulation_mode=1,
-            local_mode=False,
-        ):
-            socket_server = SocketServer(
-                host=tcpip.LOCAL_HOST, port=0, simulation_mode=1
-            )
-            await asyncio.wait_for(socket_server.start(), timeout=5)
-            port = socket_server.port
-            self.csc.port = port
-            self.assertFalse(socket_server.connected)
-            await salobj.set_summary_state(
-                remote=self.remote, state=salobj.State.ENABLED
-            )
-            self.assertTrue(socket_server.connected)
             await self.validate_telemetry()
+
             await salobj.set_summary_state(
                 remote=self.remote, state=salobj.State.DISABLED
             )
-            self.assertFalse(socket_server.connected)
-            await socket_server.exit()
+            self.assertFalse(self.socket_server.connected)
+            await self.socket_server.exit()
+
+    async def test_receive_telemetry(self):
+        logging.info("test_receive_telemetry")
+        self.disconnected_channel = None
+        self.missed_channels = 0
```

Comment:
I strongly suggest making these arguments instead of class variables. It makes the control flow more obvious: you are passing values to a method that then does some documented thing with them.

Reply:
Good suggestion. Done.